### DISCUSSION TITLE
Led Flashes

### DISCUSSION THREAD

**Discussion by byeounggwon** [awawa-dev/HyperHDR#50] [Q&A]
my product configuration.

> pi4 + hyperhdr
> hdmi grabber + hdmi splitter
> esp8266(ch340) v3 e12
> ws8212b (60/1M) 300led

There was a delay in Arduion, so I replaced it with esp8266.
The delay is gone, but the led is shaking.

The previous Arduino had a delay, but no tremors.

From the rainbow signal.
Flashes only the third blue from the left.

When I checked the voltage,
The start and end are uniform at 4.4v to 4.7v.

I also checked the resistance, level converter, and power adapter additionally, but the problem continues.

What could be the cause?

I'm sorry for my poor English.

video to the link.

[https://blog.naver.com/lomazzi/222285946655](url)

https://user-images.githubusercontent.com/74481191/112343879-e5397400-8d06-11eb-81f0-a48166bafda2.mp4

**Comment by awawa-dev** [comment]
Hi
How its wired and what level shifter do you use? Is there a common ground for all components?
Does the problem occur on static color (using effect not grabber)?

**Reply by byeounggwon** [reply]
Hi, long time no see.^^
I'm attaching an image.

Yeah, I'm shaking on a static screen.
and the ground is connected as one.

I've tested more than three level shifters.
It's not there, but it's the same.

![KakaoTalk_20210325_012911923](https://user-images.githubusercontent.com/74481191/112347112-f59f1e00-8d09-11eb-9033-230f8fba7a5a.jpg)

**Comment by awawa-dev** [comment]
Frankly I can only guess how it's connected on the right side and know nothing how it's wired to the led and the led strip ;) I think that resistor is not necessary and do you use it with arduino? BTW how many ohms? I lost ability to read the ratings from color lines years ago ;) . The level shifter you use is not recommended for high speed communication but it works in most case.
OK, but these things may or may not be related to your problem. If the flashing is limited to only one led I would suspect that led has problems on lower (from some reason) data line voltage and would replace it.

**Reply by byeounggwon** [reply]
Communicate with pi4 and esp8266 except for arduino.
It's 220 ohms.

It's 4.4~4.7v for all lines, should I raise it a little higher?

**Reply by awawa-dev** [reply]
So you have that 220ohms resistor only for esp8266 with level shifter but not for Arduino? 
The voltage is obviously not ideal but probably it would affect setup with Arduino also and/or most of the leds in the strip. But again it also depends how it all is wired. You can raise it a little but be careful because these readings probably depends how much the power supply is loaded.

**Comment by byeounggwon** [comment]
This is my current configuration.

Same configuration, no flicker in the arduino.

Flashes after replacing with esp8266....

![lbk](https://user-images.githubusercontent.com/74481191/112420241-3d589080-8d70-11eb-96fd-0f035e4ecd1c.jpg)

**Reply by awawa-dev** [reply]
Looks OK, but my comments about the level shifter you chose and that resistor are still actual.
I try to find the difference between setup with the Arduino and the Esp, it looks that not only device has changed:
- do you use the resistor with the Arduino and if not then why?
- the cable connecting device to the led strip has the same length both for Arduino and the Esp?
- do you try to use Esp without the level shifter and the resistor, connected directly to the led strip?
- _The start and end are uniform at 4.4v to 4.7v._ that means the voltage at the start is equal to the reading at the end and both are fluctuating?

There is subtle difference about how the signal is generated with the Arduino and Esp (voltage is of course different too) but it should work for both setups. The more significant difference is the data line is much more saturated if you use it at 2Mb with Esp (do you?), causing much higher refresh rate  and it's too much for the damaged led. Again I suspect defective single led and would replace it. If there had been a problem with the signal (voltage or noise) probable you would have problems with first or second led or no reaction at all from the led strip. We had similar case once: led strip was working with the Arduino and was completely refusing to cooperate with the Esp even with level shifter (first led was damaged and left in half-dead state, working only at lower refresh mode).
There is also a very, very little chance that is not the third led is damaged but the second one.

**Reply by byeounggwon** [reply]
*Resistance is not used for the arduino.
Esp8266 was not used either. I connected it later because it was blinking just in case.
*Yes, the distance between the connectors is the same for the arduino and esp8266.
*There was no resistance at first. Blinking even then
*Your start and end are the same bolts. It seems to vary from 4.4 to 4.7v depending on the color. I posted an ampere today. The bolts at the start and end have risen to 4.9v. The change is still flashing.

Flashes in whole or in part occur only in blue or mixed colors, not in particular led.

The test I did today.
>Resistance has been deleted.---Still flashing
>The first led has been replaced. ---Still flashing
>You have increased the ampere increased. ---Still flashing
>There is a led even after my monitor. I transferred and tested the esp8266 on TV without any changes, but there is no flicker. ---No blinking



1. Move the esp8266 on TV to the monitor led and attach the tested video.
   ((Only change the number of leads in hyperhdr)

https://user-images.githubusercontent.com/74481191/112500112-707c3d80-8dcb-11eb-8171-ad88d9f148cc.mp4


2. I marked the red circle on the shaking color when it was Rainbow color.
   (On the live screen, the blue area shakes and blinks.)
  
![KakaoTalk_20210326_001358079](https://user-images.githubusercontent.com/74481191/112500138-770ab500-8dcb-11eb-912a-81baa97d3f55.png)


If you don't see the video, please download it and watch it.

**Reply by awawa-dev** [reply]
I asked you before about testing on static color, no rainbow, no movie or whatever. One color red, blue or whatever set for all the led strip. What's the result? Stop testing using image from grabber before you have that results with static colors because grabber introduces some subtle shaking even on static scenes and it may be not related to the led strip at all. Simply with higher refresh rate using esp you noticed it just now. Saw the movie but it's hard to make a conclusion.

**Reply by byeounggwon** [reply]
I was busy, so I just started again. I've just started over.
We've finally solved it's over! :)

I don't know the exact cause. But the correction is...
1.hyperhdr, reinstalled wled (latest version)
2. Readjust Maximum Current in wled

Now the tremor is completely gone and it works perfectly.
There is no delay at all.

Thank you so much!!

https://user-images.githubusercontent.com/74481191/115420976-8cda9100-a236-11eb-9bd7-9046b78f31ed.mp4
